api/apps/geoprocessing/src/modules/planning-area/planning-area-tiles/planning-area-tiles.service.ts

```diff
@@ -0,0 +1,56 @@
+import { Injectable, Inject, Logger } from '@nestjs/common';
+import { ApiProperty } from '@nestjs/swagger';
+import {
+  IsUUID,
+} from 'class-validator';
+import { InjectRepository } from '@nestjs/typeorm';
+import { PlanningArea } from '@marxan/planning-area-repository/planning-area.geo.entity';
+import { TileService } from '@marxan-geoprocessing/modules/tile/tile.service';
+import { TileRequest } from '@marxan/tiles';
+import { Repository } from 'typeorm';
+export class TileSpecification extends TileRequest {
+  @ApiProperty()
+  @IsUUID()
+  planningAreaId!: string;
+}
+
+@Injectable()
+export class PlanningAreaTilesService {
+  private readonly logger = new Logger(this.constructor.name);
+
+  constructor(
+    @InjectRepository(PlanningArea)
+    private readonly repository: Repository<PlanningArea>,
+    @Inject('TileService')
+    private readonly tileService: TileService,
+  ) {}
+
+  buildCustomPlanningAreaWhereQuery(planningAreaId: string): string {
+    /**
+     * @todo this generation query is a bit...
+     */
+    let whereQuery = `project_id = '${planningAreaId}'`;
```

Comment:
well, we know that `TileService` is the headquarter of sql injections, and I won't ask to undo this in this PR, but looking at `fetchTileFromDatabase()` this shouldn't be _that_ hard. For example, this could be refactored to return a query template + parameter pair, and this would be used to split query from parameter in the `subQuery.andWhere()` where this is used.

Likewise for other parts of `fetchTileFromDatabase()` (or at the very least validating that `extent`, `buffer` and `inputProjection` are integers within an allowed range, restricting the names of the `geometry` columns to a locked-down set, and so on. But this is a story for another day, for sure.